source/blood/src/common.cpp

```diff
@@ -318,6 +318,56 @@ static void G_AddSteamPaths(const char *basepath)
     addsearchpath(buf);
 }
 
+static void ParseDesktopFile(const char *desktop)
+{
+    buildvfs_fd fd = buildvfs_open_read(desktop);
+    int32_t size = buildvfs_length(fd);
+    char *dskbufstart, *dskbuf, *dskbufend, *dskpath;
+
+    if (size <= 0)
+        return;
+
+    dskbufstart = dskbuf = (char*)Xmalloc(size);
+    size = (int32_t)buildvfs_read(fd, dskbuf, size);
+    buildvfs_close(fd);
+    dskbufend = dskbuf + size;
+
+    char buf[BMAX_PATH];
+
+    while (dskbuf < dskbufend)
+    {
+        if (Bstrncmp(dskbuf, "Path=", 5) == 0)
+        {
+            dskpath = dskbuf += 5;
+
+            while (*dskbuf++ != '\n') {}
+
+            Bsnprintf(buf, dskbuf - dskpath, dskpath);
+            Bstrcat(buf, "/data");
+
+            addsearchpath(buf);
+
+            break;
+        }
+        while (*dskbuf++ != '\n') {}
+    }
+
+    Xfree(dskbufstart);
+}
+
+
+static void G_AddGogPaths(const char *homepath)
```

Comment:
Another way of adding the gog path would be, like AddSteamPaths, add the default directory